BookOfHabits/BookOfHabitsMicroservice.Application.Service/HabitsApplicationService.cs

```diff
@@ -0,0 +1,146 @@
+﻿using AutoMapper;
+using BookOfHabitsMicroservice.Application.Models.Habit;
+using BookOfHabitsMicroservice.Application.Services.Abstractions;
+using BookOfHabitsMicroservice.Application.Services.Abstractions.Exceptions;
+using BookOfHabitsMicroservice.Application.Services.Implementations.Base;
+using BookOfHabitsMicroservice.Domain.Entity;
+using BookOfHabitsMicroservice.Domain.Entity.Enums;
+using BookOfHabitsMicroservice.Domain.Entity.Propertys;
+using BookOfHabitsMicroservice.Domain.Repository.Abstractions;
+using BookOfHabitsMicroservice.Domain.ValueObjects;
+
+namespace BookOfHabitsMicroservice.Application.Services.Implementations
+{
+    public class HabitsApplicationService(IRepository<Habit, Guid> habitRepository,
+                                          IRepository<Person, Guid> personRepository,
+                                          IRepository<Room, Guid> roomRepository,
+                                          IRepository<Delay, Guid> delayRepository,
+                                          IRepository<Repetition, Guid> repetitionRepository,
+                                          IRepository<TimeResetInterval, Guid> timeResetIntervalRepository,
+                                          IMapper mapper) : BaseService, IHabitsApplicationService
+    {
+        public async Task<HabitModel?> AddHabitAsync(CreateHabitModel cardInfo, CancellationToken token = default)
+        {
+            Person owner = await personRepository.GetByIdAsync(x => x.Id.Equals(cardInfo.PersonId), cancellationToken: token)
+                ?? throw new NotFoundException(FormatFullNotFoundErrorMessage(cardInfo.PersonId, nameof(Person)));
+
+            Room room = await roomRepository.GetByIdAsync(x => x.Id.Equals(cardInfo.RoomId), includes: $"_habits", cancellationToken: token)
+                ?? throw new NotFoundException(FormatFullNotFoundErrorMessage(cardInfo.RoomId, nameof(Room)));
+
+            var delay = new Delay(isAfterATime: false,
+                                  afterTime: 0,
+                                  isEndless: true,
+                                  durationTime: 3600);
+
+            var repetition = new Repetition(maxCountPositive: 5,
```

Comment:
we should get rid of 'magic numbers' from the our code and instead of it we can use constants